cmd/query/app/server.go

```diff
@@ -58,9 +60,10 @@ type Server struct {
 	httpConn           net.Listener
 	cmuxServer         cmux.CMux
 	grpcServer         *grpc.Server
-	httpServer         *http.Server
+	httpServer         *httpServer
 	separatePorts      bool
 	unavailableChannel chan healthcheck.Status
+	waitGroup          sync.WaitGroup
```

Comment:
nit: name fields after their purpose, not their implementation. E.g. `bgFinished`